test-series/4/tools/sim_prep.py

```diff
@@ -0,0 +1,626 @@
+    """
+    Downselect simulated array to match unique baselines from reference.
+    
+    Parameters
+    ----------
+    sim_uvd : :class:`pyuvdata.UVData`
+        :class:`pyuvdata.UVData` object containing the simulation data.
+        
+    ref_uvd : :class:`pyuvdata.UVData`
+        :class:`pyuvdata.UVData` object containing the reference data. 
+        Only needs to have the metadata loaded.
+        
+    tol : float, optional
+        The maximum allowable discrepancy in antenna position components.
+        Default is 1 meter.
+        
+    Returns
+    -------
+    new_sim_uvd : :class:`pyuvdata.UVData`
+        Simulation data adjusted to contain a subset of the simulated array
+        such that, after appropriately translating the antenna positions, 
+        the maximum number of antennas/unique baselines remain in the 
+        intersection of the simulation subarray and the reference array. 
+        The relevant attributes of the simulation data are updated 
+        accordingly.
+
+    Notes
+    -----
+    See the memo "Antennas, Baselines, and Maximum Overlap" for details on 
+    the antenna selection process. It is worth noting that the algorithm 
+    implemented here is *not* appropriate for general arrays, but the 
+    general case should reduce to this case for an array with appropriate 
+    symmetry, as we have for the RIMEz simulation.
+    """
+    sim_uvd = _sim_to_uvd(sim_uvd)
+
+    # Find the optimal intersection, get the map between antennas, and downselect the data.
+    sim_ENU_antpos = _get_antpos(sim_uvd)
+    ref_ENU_antpos = _get_antpos(ref_uvd)
+    array_intersection = _get_array_intersection(sim_ENU_antpos, ref_ENU_antpos, tol)
+    sim_to_ref_ant_map = _get_antenna_map(array_intersection, ref_ENU_antpos, tol)
+    sim_uvd.select(antenna_nums=list(sim_to_ref_ant_map.keys()), keep_all_metadata=False)
+    ref_uvd.select(antenna_nums=list(sim_to_ref_ant_map.values()), keep_all_metadata=False)
+    
+    # Prepare the new data array.
+    new_sim_data = np.zeros_like(sim_uvd.data_array, dtype=np.complex)
+    for antpairpol in sim_uvd.get_antpairpols():
+        ai, aj, pol = antpairpol
+        ref_antpairpol = (sim_to_ref_ant_map[ai], sim_to_ref_ant_map[aj], pol)
+        blts, conj_blts, pol_inds = ref_uvd._key2inds(ref_antpairpol)
+        sim_data = sim_uvd.get_data(antpairpol)
+        if len(blts) > 0:
+            this_slice = (blts, 0, slice(None), pol_inds[0])
+        else:
+            this_slice = (conj_blts, 0, slice(None), pol_inds[1])
+            sim_data = sim_data.conj()
+        new_sim_data[this_slice] = sim_data
+        
+    # Actually update the data and metadata.
+    sim_uvd.ant_1_array = ref_uvd.ant_1_array
+    sim_uvd.ant_2_array = ref_uvd.ant_2_array
+    sim_uvd.antenna_numbers = ref_uvd.antenna_numbers
+    sim_uvd.antenna_positions = ref_uvd.antenna_positions
+    sim_uvd.telescope_location = ref_uvd.telescope_location
```

Comment:
I'm not too sure what the exact argument for this is to be honest--I'm only aware of the argument that there has been difficulty running redcal/abscal on simulation data before, so the safe route to take is to make sure all of the telescope-related information matches the observational metadata.

Reply:
@jsdillon can you weigh in on this?

Reply:
I added some clarifying comments, but I'm going to hold off on marking this as resolved--I'd like confirmation that the explanation I provided is OK.

Reply:
I guess I meant that it seems unintuitive to set the simulated "ideal-position" baselines to the "non-ideal" baselines of the data, since those baselines don't exactly match what was simulated. But IIRC @jsdillon was saying that the calibration pipeline never actually uses the real baselines, rather it just uses them to get the idealized positions and uses those. So in fact setting them to the non-ideal positions lets that assumption be tested a bit as well.